interceptor/forward_wait_func.go

```diff
@@ -1,26 +1,36 @@
 package main
 
 import (
+	"errors"
 	"fmt"
 	"log"
 	"time"
 
+	"github.com/go-logr/logr"
 	"github.com/kedacore/http-add-on/pkg/k8s"
 	appsv1 "k8s.io/api/apps/v1"
 )
 
 type forwardWaitFunc func() error
 
 func newDeployReplicasForwardWaitFunc(
+	logger logr.Logger,
 	deployCache k8s.DeploymentCache,
 	deployName string,
 	totalWait time.Duration,
 ) forwardWaitFunc {
+	logger = logger.WithName("newDeployReplicasForwardWaitFunc")
 	return func() error {
 		deployment, err := deployCache.Get(deployName)
 		if err != nil {
 			// if we didn't get the initial deployment state, bail out
-			return fmt.Errorf("Error getting state for deployment %s (%s)", deployName, err)
+			logger.Error(
```

Comment:
NOTE: this block is significantly changed in https://github.com/kedacore/http-add-on/pull/181/files. The conflict should be resolved favoring the other PR